### PR TITLE
Adding loader for Beatport EDM key

### PR DESCRIPTION
**Pull request by PRamoneda**
### Adding loader for Beatport EDM key

 Please use the following title: "Adding loader for MyDATASET". If your pull request is work in progress, change your title to "[WIP] Adding loader for MyDATASET" to avoid reviews while the loader is not ready.

### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/datasets.rst`
- [x] Add the module to `mirdata/__init__.py`
- [x] Add the module to the list in the `README.md` file, section `Currently supported datasets`

If your dataset **is not fully downloadable** there are two extra steps you should follow:
It is fully downloadable

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

### CODE REVIEW

File: mirdata/beatport_key.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        remotes=REMOTES,
+        partial_download=partial_download,
+        info_message=download_message,
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+    # This dataset annotations have characters that doesnt correspond with json format. In particular, "bpm": nan
+    # doesn't correspond to json format.
+    # input file
+    find_replace(os.path.join(data_home, "meta"), ": nan", ": null", "*.json")
```
Comment (nkundiushuti) [inline mirdata/beatport_key.py:197]: what is the role of this function when you have an index for all the files in the dataset?

File: mirdata/jams_utils.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def jams_converter(
 
 
 def beats_to_jams(beats):
```
Comment (nkundiushuti) [inline mirdata/jams_utils.py:251]: this file should not be included in this PR

File: mirdata/rwc_classical.py [hunk 3]
```diff
@@ -287,14 +287,14 @@ def track_ids():
 def load(data_home=None):
```
Comment (nkundiushuti) [inline mirdata/rwc_classical.py:287]: this file should not be included in this PR

File: mirdata/utils.py [hunk 4]
```diff
@@ -208,10 +208,10 @@ def load_json_index(filename):
```
Comment (nkundiushuti) [inline mirdata/utils.py:208]: this file should not be included in this PR

File: mirdata/beatport_key.py (hunk 1 above)
Comment (PRamoneda) [reply mirdata/beatport_key.py:197]: Original JSON files have `nan` that are not of the JSON standard format. So I can't read JSON. So from what is downloaded of Zenodo is changed each `nan` to `none`.

File: mirdata/jams_utils.py (hunk 2 above)
Comment (PRamoneda) [reply mirdata/jams_utils.py:251]: Black modifies it.

File: mirdata/rwc_classical.py (hunk 3 above)
Comment (PRamoneda) [reply mirdata/rwc_classical.py:287]: Black modifies it.

File: mirdata/utils.py (hunk 4 above)
Comment (PRamoneda) [reply mirdata/utils.py:208]: Black modifies it.

File: mirdata/beatport_key.py [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+        self.metadata_path = (
+            os.path.join(self._data_home, self._track_paths['meta'][0])
+            if self._track_paths['meta'][0] is not None
+            else None
+        )
+        self.title = self.audio_path.replace(".mp3", '').split('/')[-1]
+
+    @utils.cached_property
+    def key(self):
+        """String: key annotation"""
```
Comment (magdalenafuentes) [inline mirdata/beatport_key.py:93]: I've noticed we're mixing docs styles. I don't know if this is the only loader that does it, but I'll add it as part of the audit issue to make sure we are consistent later on

File: docs/source/datasets.rst [hunk 6]
```diff
@@ -30,6 +30,9 @@ click the link provided in the "Module" column.
 |                  | | Dataset           | - annotations: ✅   | - :ref:`sections`         |        |
 |                  |                     |                     | - :ref:`key`              |        |
 +------------------+---------------------+---------------------+---------------------------+--------+
+| giantsteps_key_  | Giantsteps EDM key  | - audio: ✅         | - global :ref:`key`       | 1486   |
+|                  |                     | - annotations: ✅   |                           |        |
```
[the remark this answers is not in the document]
Comment (magdalenafuentes) [reply docs/source/datasets.rst:34]: Is the audio provided? or should it be an X?

File: mirdata/beatport_key.py (hunk 1 above)
Comment (magdalenafuentes) [reply mirdata/beatport_key.py:197]: I understand why the function is there now. Maybe it would be more informative adding the long comment on the readme of the function above and leaving a short one here. Also, thinking out loud, I guess is a good choice to do it in the download so you do it only once and in a consistent way to the index, right?

File: mirdata/jams_utils.py (hunk 2 above)
Comment (magdalenafuentes) [reply mirdata/jams_utils.py:251]: Try running black -S

File: tests/resources/mir_datasets/beatport_key/meta/100066 Lindstrom - Monsteer (Original Mix).json [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+   "height": 250
+  }, 
+  "dynamic": {
+   "url": "https://geo-media.beatport.com/image_size/{w}x{h}/2b8bd856-55ad-40fc-b754-5bca7d219c0c.png", 
+   "id": 2422005
+  }
+ }, 
+ "guest_pick": false, 
+ "sub_genres": []
+}
```
Comment (magdalenafuentes) [inline tests/resources/mir_datasets/beatport_key/meta/100066 Lindstrom - Monsteer (Original Mix).json:127]: nit: end line

File: tests/test_beatport_key.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        'genres': dict,
+        'artists': list,
+        'tempo': int
+    }
+
+    run_track_tests(track, expected_attributes, expected_property_types)
+
+    audio, sr = track.audio
+    assert sr == 44100, 'sample rate {} is not 44100'.format(sr)
+    assert audio.shape == (5292000,), 'audio shape {} was not (5294592,)'.format(
```
Comment (magdalenafuentes) [inline tests/test_beatport_key.py:33]: shouldn't numbers match here?

File: mirdata/beatport_key.py [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=None, mono=True)
+
+
+def find_replace(directory, find, replace, pattern):
+    """
+    Replace in some directory all the songs with the format pattern find by replace
+    Parameters
```
[the remark this answers is not in the document]
Comment (magdalenafuentes) [reply mirdata/beatport_key.py:146]: See comment below

**Review by magdalenafuentes — CHANGES_REQUESTED** [review]

**Comment by magdalenafuentes** [comment]
@PRamoneda great job! Left my comments there. After you address them I'll take a look again, but first maybe it would be good to also merge the master so I can check and merge right away?

File: mirdata/beatport_key.py (hunk 1 above)
Comment (PRamoneda) [reply mirdata/beatport_key.py:197]: yes, right

File: docs/source/datasets.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (PRamoneda) [reply docs/source/datasets.rst]: done!!!
yes!! the audio is provided!!

File: mirdata/beatport_key.py [hunk 10]
```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+"""beatport_key Dataset Loader
+The Beatport EDM Key Dataset includes 1,486 two-minute sound excerpts from various EDM
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/beatport_key.py:3]: ok perfect!
done

File: mirdata/beatport_key.py [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+
+Ángel Faraldo (2017). Tonality Estimation in Electronic Dance Music: A Computational and Musically Informed
+Examination. PhD Thesis. Universitat Pompeu Fabra, Barcelona.
+
+This dataset is mainly intended to assess the performance of computational key estimation algorithms in electronic
+dance music subgenres.
+
+Data License: Creative Commons Attribution Share Alike 4.0 International
+"""
+import fnmatch
```
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/beatport_key.py:20]: I think that fnmatch is from "python standard"

File: mirdata/beatport_key.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (PRamoneda) [reply mirdata/beatport_key.py:146]: ok! done then

File: mirdata/beatport_key.py (hunk 5 above)
Comment (PRamoneda) [reply mirdata/beatport_key.py:93]: I was imitating beatles loader. What style must we use?

File: tests/test_beatport_key.py (hunk 8 above)
Comment (PRamoneda) [reply tests/test_beatport_key.py]: done

**Comment by PRamoneda** [comment]
@magdalenafuentes I dont know I have requested or not the review! hahahaa Thnks so much!

**Review by magdalenafuentes — APPROVED** [review]

File: mirdata/rwc_classical.py (hunk 3 above)
Comment (nkundiushuti) [reply mirdata/rwc_classical.py]: try running it as defined in CONTRIBUTING
`black --target-version py37 --skip-string-normalization mirdata/`
as a rule of thumb try avoiding global commits and refer to specific files.
you can discard the commits on a specific file by updating the master and replacing your version of the file with the version from the master 
`git checkout master file_to_restore.ext`
